day_7/index.js

```diff
@@ -1,320 +1,543 @@
-        if (arr2[j] === 10) {
-            arr2[j] = 'a';
-        } else if (arr2[j] === 11) {
-            arr2[j] = 'b';
-        } else if (arr2[j] === 12) {
-            arr2[j] = 'c';
-        } else if (arr2[j] === 13) {
-            arr2[j] = 'd';
-        } else if (arr2[j] === 14) {
-            arr2[j] = 'e';
-        } else if (arr2[j] === 15) {
-            arr2[j] = 'f';
-        }
-    }
-    arr3[0] = String(arr2[0]) + String(arr2[1]) + String(arr2[2]) + String(arr2[3]) + String(arr2[4]) + String(arr2[5]);
-
-    return console.log(arr3);
-
-}
-
-
-const arrayOfHexaColors2 = (count) => {
-    let hexa = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 'a', 'b', 'c', 'd', 'e', 'f'];
-    let array = [];
-
-
-    for (i = 0; i <= count; i++) {
-        let str = '';
-        for (j = 0; j <= 6; j++) {
-            str += hexa[Math.floor(Math.random() * 15 + 1)];
-        }
-        array.push('#' + str);
-
-    }
-
-    return console.log(array);
-}
-
-const generateColors = (numberName, count) => {
-    if (numberName == 'hex') {
-        arrayOfHexaColors2(count);
-    } else if (numberName == 'rgb') {
-        rgbColorGenerator(count);
-
-    }
-
-}
-
-generateColors('rgb', 2);
\ No newline at end of file
+const fullName = (firstName = 'Артур', lastName = 'Москвичёв') => {
+  let space = ' ';
+  let fullName = firstName + space + lastName;
+  return console.log(fullName);
+};
+fullName('Денис', 'Токарев');
+
+const addNumbers = (firstNumber, secondNumber) => {
+  let sum = firstNumber + secondNumber;
+  return console.log(sum);
+};
+
+addNumbers(1, 2);
+
+const areaOfRectangle = (length, width) => {
+  let area = length * width;
+  return console.log(area);
+};
+
+areaOfRectangle(10, 8);
+
+const perimeterOfRectangle = (length, width) => {
+  let perimeter = 2 * (length + width);
+  return console.log(perimeter);
+};
+
+perimeterOfRectangle(10, 8);
+
+const volumeOfRectPrism = (length, width, heigth) => {
+  let volume = length * width * heigth;
+  return console.log(volume);
+};
+
+volumeOfRectPrism(10, 12, 13);
+
+const areaOfCircle = (radius) => {
+  let areaCircle = Math.PI * radius * radius;
+  return console.log(areaCircle);
+};
+
+areaOfCircle(10);
+
+const lengthOfCircle = (radius) => {
+  let lengthCircle = 2 * Math.PI * radius;
+  return console.log(lengthCircle);
+};
+
+lengthOfCircle(10);
+
+const density = (weight, volume) => {
+  let Density = weight / volume;
+  return console.log(Density);
+};
+
+density(10, 12);
+
+const speed = (distance, time) => {
+  let speed = distance / time;
+  return console.log(speed);
+};
+
+speed(100, 2);
+
+const ITM = (weight, heigth) => {
+  let itm = weight / (heigth * heigth);
+  if (itm < 18.5) {
+    return console.log('Недостаточный вес');
+  } else if (itm >= 18.5 && itm <= 24.9) {
+    return console.log('Нормальный вес');
+  } else if (itm >= 25 && itm <= 29.9) {
+    return console.log('Избыточный вес');
+  } else {
+    return console.log('Ожирение');
+  }
```

Comment:
`return` можно использовать здесь для избавления от вложенности
```suggestion
  if (itm < 18.5) {
    return console.log('Недостаточный вес');
  } 
  
  if (itm <= 24.9) {
    return console.log('Нормальный вес');
  } 
  
  if (itm <= 29.9) {
    return console.log('Избыточный вес');
  } 
  
  return console.log('Ожирение');
```

Reply:
Почему я доебался до `return console.log`, а сам так делаю в моем примере? Тут немного другое, смотри:
* `console.log` – функция, не возвращающая результат (тип результата – `undefined`)
*  `return` берет данное ему значение и возвращает его из функции, прекращая ее выполнение

В твоих вариантах использования смысла в применении `return` и `console.log` вместе нет – выполнение тела функции и так завершилось бы после `console.log`, а `return`'y возвращать нечего, потому что `console.log` возвращает `undefined`.

В моих примерах происходит то же самое, но `return` имеет смысл, т.к. да, возвращать нечего, но мы просто выходим из функции, и в нашем случае это просто более короткий способ записать `console.log(); return;`. Если мы выходим из функции внутри некоторых `if`, это гарантирует, что:
* если условие внутри if правдиво, мы выполним то, что внутри {} и закончим выполнять функцию
* если оно ложно, мы продолжим выполнять то, что идет после `}`
*Таким образом, мы можем использовать это, чтобы превратить `if () { ... } else if () { ... } ... else if () { ... } else { ... }` в `if () { ...; return; } if () { ...; return; } ... return;` и else нам по сути не нужен, т.к. уже гарантировано, что выполнение дальше не пойдет

Reply:
@devlato  спасибо за расширенный ответ!)